### DISCUSSION TITLE
moose phase field

### DISCUSSION THREAD

**Discussion by Adairle** [idaholab/moose#22666] [Q&A Modules: Phase field] [answered]
Holle everyone
Excuse me, the old phase field model program reported an error on the new server
How to solve？ thanks
![$A@B`MBSZJ%%VQ6X@0PJ0XE](https://user-images.githubusercontent.com/115397737/201290069-35360da0-78d6-4190-ac50-384321ac05c0.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

It's as the error message says, the program you are using is using deprecated constructs and can no longer build.
See this post for more details
https://mooseframework.inl.gov/newsletter/2021_11.html

Guillaume

**Reply by Adairle** [reply]
Thank you for your help. I'll operate it now

**Reply by Adairle** [reply]
Hello

How can the source file and header file of the old version apply to the new version of moose？
Or is there any way to download the old version of Moose?
thanks！

**Reply by GiudGiud** [reply]
You need to remove the old validParams logic and use the new syntax, as the newsletter says

**Reply by GiudGiud** [reply]
We would rather you update the app you are using (which one is it?) than use an old version of MOOSE

**Reply by Adairle** [reply]
The latest system in use today
Thanks again for your reply！
